pkg/webhook/cluster/mutation/mutation.go

```diff
@@ -196,31 +185,6 @@ func (h *AdmissionHandler) mutateUpdate(oldCluster, newCluster *kubermaticv1.Clu
 		}
 	}
 
-	// For backward compatibility, if CNIPlugin is not set (possible only for clusters started before KKP v2.18 release),
-	// explicitly set it to the last Canal CNI version supported by KKP before v2.18.
-	if newCluster.Spec.CNIPlugin == nil {
-		newCluster.Spec.CNIPlugin = &kubermaticv1.CNIPluginSettings{
-			Type:    kubermaticv1.CNIPluginTypeCanal,
-			Version: cni.CanalCNILastUnspecifiedVersion,
-		}
-	}
-	// This part handles CNI upgrade from unsupported CNI version to the default Canal version.
-	// This upgrade is necessary for k8s versions >= 1.22, where v1beta1 CRDs used in old Canal version (v3.8)
-	// are not supported anymore.
-	if newCluster.Spec.CNIPlugin.Type == kubermaticv1.CNIPluginTypeCanal &&
```

Comment:
I think we should still keep this part to support upgrade of k8s versions >= 1.22. In KKP 2.19 all clusters that previously had `cluster.Spec.CNIPlugin == nil` now have CNI set to `canal` of version `v3.8`.